### PR TITLE
add search folder methods

### PR DESCRIPTION
**Pull request by jackneary**
added text box search and snippet searches through a given folder

### CODE REVIEW

**Comment by jackneary** [comment]
@justinnhli @crzeller11

File: search.py [hunk 1]
```diff
@@ -283,3 +316,51 @@ def search(input_root, data_tree):
                 # elements don't match->stop input iteration and move to next data element
                 break
     return measure_match_list
+
```
Comment (justinnhli) [inline search.py:322]: All lint output for the lines below (l321-l366) needs to be fixed.

File: search.py [hunk 2]
```diff
@@ -283,3 +316,51 @@ def search(input_root, data_tree):
...
+    Returns:    text_box_array [List<string>]: list of the path of the file that contains the given tag and the measure
+                                                in which it is found
+    """
+    file_list = get_mei_from_folder(path)
+    text_box_array = []
+    for file in file_list:
+        _, root = prepare_tree(file)
+        list = text_box_search(root, tag, search_term)
+        string_list = []
+        if len(list) != 0:
```
Comment (justinnhli) [inline search.py]: The if is unnecessary - if the list has 0 length, range will already return the empty sequence.

File: search.py [hunk 3]
```diff
@@ -283,3 +316,51 @@ def search(input_root, data_tree):
...
+                input_root is a root of elements to be searched for
+    Returns:    regular_search_array[List<sring>]: list of the path and measure number in which the snippet is found
+    """
+    file_list = get_mei_from_folder(path)
+    regular_search_array = []
+    for file in file_list:
+        tree, _ = prepare_tree(file)
+        list = search(input_root, tree)
+        string_list = []
+        if len(list) != 0:
```
Comment (justinnhli) [inline search.py]: Ditto

File: tests.py [hunk 4]
```diff
@@ -61,16 +65,41 @@ def positive_test_check_element_match():
             "check_element_match: element not equal to themselves; check Element with id " + element.attrib["xml:id"]
 
 
+def positive_test_text_box_search_folder():
```
Comment (justinnhli) [inline tests.py:66]: lint output for the lines below needs to be fixed.

File: search.py (hunk 1 above)
Comment (jackneary) [reply search.py:322]: are there any issues in particular that you feel need to be changed before you can accept?

Comment (jackneary) [reply search.py:322]: @justinnhli

Comment (justinnhli) [reply search.py:322]: Not outside of what I commented on. You added test cases for the new functions, so that's good.

File: search.py [hunk 5]
```diff
@@ -283,3 +316,49 @@ def search(input_root, data_tree):
...
+                tag [string]: element type
+                search_term[string]: search term to find element
+    Returns:    text_box_array [List<string>]: list of the path of the file that contains the given tag and the measure
+                                                in which it is found
+    """
+    file_list = get_mei_from_folder(path)
+    text_box_array = []
+    for file in file_list:
+        _, root = prepare_tree(file)
+        array = text_box_search(root, tag, search_term)
```
Comment (justinnhli) [inline search.py]: This is a better name, but still not indicative. An array of what? A much better name would be `search_results`

**Comment by justinnhli** [comment]
Also, please rebase on master before continuing.

**Comment by jackneary** [comment]
@justinnhli I think I have done all linting for the new code in this pull request

**Pull request merged**